crates/opencascade/examples/chamfer.rs

```diff
@@ -0,0 +1,47 @@
+use glam::dvec3;
+use opencascade::{
+    primitives::{Edge, Face, Solid, ToAngle},
+    workplane::Workplane,
+};
+
+pub fn main() {
+    let mut base = Workplane::xy().rect(10.0, 10.0);
+    // asymmetric chamfer
+    base.chamfer(1.0, Some(2.0));
+
+    let mut top = Workplane::xy().rect(10.0, 10.0);
+    // 45 degree chamfer - other angles will be asymmetric
+    top.chamfer_angle(2.0, 45.degrees());
+    top.translate(dvec3(0.0, 0.0, 10.0));
+
+    let chamfered_box = Solid::loft([&base, &top].into_iter());
+
+    let handle = Workplane::xy().rect(5.0, 5.0);
+    let handle_face = Face::from_wire(&handle);
+
+    let handle_body = handle_face.extrude(dvec3(0.0, 0.0, -10.0));
+    let mut chamfered_shape = chamfered_box.union(&handle_body);
+
+    // chamfer the top of the protrusion by getting shape edges at z = -10.0
+    let top_edges = chamfered_shape
+        .edges()
+        .filter(|edge| edge.start_point().z == -10.0 && edge.end_point().z == -10.0)
+        .collect::<Vec<Edge>>();
```

Comment:
In general it's good to avoid direct floating-point comparisons as they can be inaccurate. If you debug print more complicated points, you'll notice a lot of them are _close_ to 0.0, but not exactly 0.0. So most operations involving floating-point comparisons will use a "tolerance" value or threshold to see if they're "close enough".

But in this case we can go with a different approach:

The "query" API of this crate is not very fleshed out yet, but we do have the ability to search for faces that are at the extremes of certain distances. It's used in the keycap example, and using it here would look like this:

```rust
let top_edges: Vec<_> = chamfered_shape
        .faces()
        .farthest(Direction::NegZ) // Get the face whose center of mass is the farthest in the negative Z direction
        .expect("Should have a face on the bottom of the handle")
        .edges() // Get all the edges of this face
        .collect();
```

You'll notice the `edges()` function isn't yet implemented for `Face`, but it's quite simple:

```rust
pub fn edges(&self) -> EdgeIterator {
    let explorer =
        TopExp_Explorer_ctor(cast_face_to_shape(&self.inner), TopAbs_ShapeEnum::TopAbs_EDGE);

    EdgeIterator { explorer }
}
```